TERMS.md

```diff
@@ -0,0 +1,121 @@
+## Service
+	- (General) A service is a software functionality or set of functionalities that can be activated by a client. A key characteristic of a service in support of SOA is that it has low coupling to other services and strives to provide a useful function also if used independently from other services.
```

Comment:
We should better align indentation to get uniform rendering. I.e. remove spaces before the dash (`-`) to get same rendering as later in the document.

Compare with the two examples below:

    - ddd
    - ddd

- rrr
- rrr

Reply:
> We should better align indentation to get uniform rendering. I.e. remove spaces before the dash (-) to get same rendering as later in the document.

I'm OK with that Erik.  But the purpose of this was first all to see if we agree with what things are, then we can work on formatting.  I don't think we will even have a document called "Terms.md" in the end.  There are various ideas we can discuss on how to provide the official documentation.

> Do we need to be so restrictive that we say "that can be activated by a client". Theoretically we can have a service that does not even offer any methods that can be called by clients, or? 

At first I was inclined to agree with this.  I've mostly tried to get the "overall picture" here so that we can find a reasonable consensus.

...but after a while I'm not sure what I think.  Does your example depend on the definition of _client_?  Does a seat belt reminder at least have a `start()` method, that will called by _something_ in the system (could be an "internal" client)? Is it likely to have some kind of settings/API/something that can be modified at least?   

Maybe I should be clear (if not obvious) that although we often talk about RPC, my mindset is basically that _any_ type of API/programming interface might be possible to cover by the IDL (it remains to see what it is used for, but it _could_ be used for anything I think).  So I'm not only considering "external" clients or anything like that.

I also think that describing concepts in a general Server/Client fashion helps understanding for most.  It might not fit all systems _exactly_ but on the other hand I don't think it will in reality put limits on the more uncommon cases?

My gut feeling is that we might benefit from putting some constraints on the term Service, for example that it must have a defined API to be called a Service.   And then decide to call other things that don't fit in, something else.  Below the service concept I imagine that any system is likely to use other lower-level terms/concepts, such as maybe "Software Component" that would cover the case you mention, if **Service** didn't.

Another way to think about it.  We are still keeping the language definition open for changes, but if I consider what we have now -> what would a definition file contain for a service that does not have an API.  I feel like it's likely to be an empty file, considering what we have now?  There's no reason for "namespaces" if you're not putting methods and such into that context, and datatypes only make sense to define in the context of an API, etc.    So if it's true that would be an empty file then it seems not such a relevant case to work on right now.

Maybe you mean it has an API but are you think it doesn't have a client (according to some definition of client?)

It might be easiest to discuss if we have counter-proposal for new text, I think.  But if you meant to simply remove it we would have: 
```
- (General) A service is a software functionality or set of functionalities.  A key characteristic of a service in support of SOA....
```

For me, the first sentence is then weak and not saying much.  We might perhaps rather rather define a "software component" that is an executable piece of software, with or without an actual API.

The rest of the text (low coupling, and in particular "SOA") suggests interaction with other components, so if there's no interface to speak of that text feels like that text also would not fit in.   We would need alternative proposal I think?

Reply:
Maybe something like below to show that they typically have an interface, but not necessarily

`(General) A service is a software functionality or set of functionalities. It typically has an interface that clients can use to interact with the service.`

Reply:
What does a service without an interface look like?

If we can imagine such a "software functionality" that has no interaction with other software parts, I would still rather call that a "Software Component" than a Service.

Reply:
Revisiting:

> Theoretically we can have a service that does not even offer any methods that can be called by clients, or? Like a seat belt reminder function that is "always on" and just starts a buzzer if anyone is not using a seat belt. Could it be interesting for us to also support "empty service declarations", i.e. only specifying name but no interfaces/methods. 

I think I'm still not fully clear on the case we're discussing to cover here.  First, my intention with "client" is very broad.  Anything that calls the interface.   So if there is an interface, there's presumably some entity using it.  

But what you're describing here is a component with no interface and method.  A completely empty component with no interface at all, I think won't be covered by the core IDL (I standing for Interface).  

I think the conclusion might be that the concept of Service shouldn't be part of the core IDL (since it is a pure interface description).  However, it still is useful that we agree on the terminology for both things in-scope and outside scope of the IDL, and some of the things might be in-scope for Layers that are defined within the full scope of this.

Furthermore, if something similar to "Franca-plus" is created, then there would be concepts there to define SW components (and perhaps "services"?), and so you could create a component without assigning any interface to it.  If something like that gets added and is usable also for non-services, then there should not be a major problem with putting a scope limit on the definition of a Service, because there ought to be other terms to define interface-less components.

> how do we then see the relationship between services and software components? Do see we that a service always belongs to (or is offered by) a software component?

Defining that relationship is probably within the "*-plus" scope talked about above.   Maximum flexibility on that level of system definition, might be to allow a service to be offered by a single, or multiple SW components that are grouped together.